### PR TITLE
jormungandr: parameter used to format bikeDetails of the request

### PR DESCRIPTION
**Pull request by kadhikari**
Ticket: https://jira.kisio.org/browse/NAVP-950

### CODE REVIEW

**Comment by xlqian** [comment]
can you add some integration tests?

File: source/jormungandr/jormungandr/street_network/geovelo.py [hunk 1]
```diff
@@ -79,27 +79,35 @@ def _pt_object_summary_isochrone(cls, pt_object):
         return [coord.lat, coord.lon, getattr(pt_object, 'uri', None)]
 
     @classmethod
-    def _make_request_arguments_isochrone(cls, origins, destinations):
+    def _make_request_arguments_bike_details(cls, bike_speed):
```
Comment (xlqian) [inline source/jormungandr/jormungandr/street_network/geovelo.py]: you can rename `bike_speed` to `bike_speed_mps`

File: source/jormungandr/jormungandr/utils.py [hunk 2]
```diff
@@ -648,3 +648,7 @@ def portable_min(*args, **kwargs):
             raise
     if PY3:
         return min(*args, **kwargs)
+
+
+def mps_to_kmph(speed):
```
Comment (xlqian) [inline source/jormungandr/jormungandr/utils.py:653]: have a look at [pint](https://github.com/hgrecco/pint)

```python

import pint
units = pint.UnitRegistry()

speed = (3.6*units.kilometer) / (2*units.second)
print(speed.to_base_unit())
print(speed.to_base_unit().magnitude)

```

File: source/jormungandr/jormungandr/street_network/geovelo.py [hunk 3]
```diff
@@ -254,7 +261,7 @@ def _get_response(cls, json_response, pt_object_origin, pt_object_destination, f
                 path_item = section.street_network.path_items.add()
                 path_item.name = geovelo_instruction[1]
                 path_item.length = geovelo_instruction[2]
-                path_item.duration = round(path_item.length * speed)
+                path_item.duration = round(path_item.length / speed if speed != 0 else 0)
```
Comment (xlqian) [inline source/jormungandr/jormungandr/street_network/geovelo.py]: round(path_item.length / speed) if speed != 0 else 0

File: source/jormungandr/jormungandr/utils.py [hunk 4]
```diff
@@ -648,3 +648,7 @@ def portable_min(*args, **kwargs):
...
+    return int(round(3.6 * speed))
```
Comment (xlqian) [inline source/jormungandr/jormungandr/utils.py]: int?

File: source/jormungandr/jormungandr/street_network/geovelo.py [hunk 5]
```diff
@@ -254,7 +261,7 @@ def _get_response(cls, json_response, pt_object_origin, pt_object_destination, f
                 path_item = section.street_network.path_items.add()
                 path_item.name = geovelo_instruction[1]
                 path_item.length = geovelo_instruction[2]
-                path_item.duration = round(path_item.length * speed)
+                path_item.duration = round(path_item.length / speed) if speed != 0 else 0
```
Comment (nonifier) [inline source/jormungandr/jormungandr/street_network/geovelo.py:264]: :scream: 
well spoted !

**Review by nonifier — APPROVED** [review]

File: source/jormungandr/jormungandr/street_network/geovelo.py [hunk 6]
```diff
@@ -79,27 +79,35 @@ def _pt_object_summary_isochrone(cls, pt_object):
         return [coord.lat, coord.lon, getattr(pt_object, 'uri', None)]
 
     @classmethod
-    def _make_request_arguments_isochrone(cls, origins, destinations):
+    def _make_request_arguments_bike_details(cls, bike_speed_mps):
+        bike_speed = mps_to_kmph(bike_speed_mps)
+
+        return {
+            'profile': 'MEDIAN',  # can be BEGINNER, EXPERT
```
Comment (nonifier) [inline source/jormungandr/jormungandr/street_network/geovelo.py:86]: Can't we use enum here to define the profile and bike type by any chance ?

Comment (pbougue) [reply source/jormungandr/jormungandr/street_network/geovelo.py:86]: Not sure it's worth it as there is no way to manage it from Navitia's API.

File: source/jormungandr/jormungandr/utils.py (hunk 2 above)
Comment (pbougue) [reply source/jormungandr/jormungandr/utils.py:653]: Fun but a bit overkill here @xlqian :smile:

**Review by pbougue — APPROVED** [review]
We could also add `"e-bike": false` but it would be more documenting geovelo's API than really useful.

**Pull request merged**